pkg/sql/sqlbase/rowwriter.go

```diff
@@ -714,7 +714,7 @@ func (ru *RowUpdater) UpdateRow(
 
 		ru.Fks.addCheckForIndex(ru.Helper.TableDesc.PrimaryIndex.ID)
 		for i := range newSecondaryIndexEntries {
-			if !bytes.Equal(newSecondaryIndexEntries[i].Key, secondaryIndexEntries[i].Key) {
+			if !bytes.Equal(newSecondaryIndexEntries[i].Key, oldSecondaryIndexEntries[i].Key) {
 				ru.Fks.addCheckForIndex(ru.Helper.Indexes[i].ID)
```

Comment:
I'm pretty sure this is going to cause a panic on an out of index

Reply:
I also checked you do not need to call ru.Fks.addCheckForIndex() for inverted indexes

Reply:
That being said it would be nice to create a schema with a FK relationship between table A -> B say on a column like ID with an index on ID and with B having another json column on it. And running through some CASCADE ON DELETE/UPDATE to make sure that it doesn't mess up.

Reply:
how can you guarantee that the index i is not gonna cause a panic when used with ru.Helper.Indexes[i]